constraint/pkg/client/drivers/k8scel/driver.go

```diff
@@ -168,11 +190,29 @@ func (d *Driver) Query(ctx context.Context, target string, constraints []*unstru
 	for _, constraint := range constraints {
 		evalStartTime := time.Now()
 		// template name is the lowercase of its kind
-		validator := d.validators[strings.ToLower(constraint.GetKind())]
-		if validator == nil {
+		wrappedValidator := d.validators[strings.ToLower(constraint.GetKind())]
+		if wrappedValidator == nil {
 			return nil, fmt.Errorf("unknown constraint template validator: %s", constraint.GetKind())
 		}
 
+		assumeVAPEnforcementNotDisabled, err := assumeVAPEnforcementWithDefault(constraint, VAPDefaultYes)
```

Comment:
Why is this defaulted to Yes when this comment above seems to indicate when no label is provided, vap generation should default to no/disabled

> If no value is provided, VAP generation
// is presumed to be disabled and the engine will
// validate ALL policies.

Reply:
This replicates the logic Gatekeeper uses. Unwinding the whole decision tree yields:

* If constraint template is undefined, use global default
* If constraint template is disabled, disable
* If constraint template is enabled, fall back to constraint logic
* If constraint is disabled, disable
* If constraint is enabled, enable so long as constraint template is enabled
* If constraint template is undefined, use template value.

Because the if statement is <if constraint enabled> && <if template enabled>, defaulting to true for the constraint means the constraint template determines the outcome.

Here is G8r's way of expressing the the same thing (note the explicit test for "No"):

https://github.com/open-policy-agent/gatekeeper/blob/45e4552595d5fcaf6f66c3d4335aea2c00973760/pkg/controller/constraint/constraint_controller.go#L309-L321

Reply:
To match the same behavior as GK, if constraint is not set to disable, this should also return VAPDefaultYes to use template value?
https://github.com/open-policy-agent/frameworks/pull/400/files#diff-5e2ac34b20b681ef5a8747fac1f9563e8c60060f9609e7ee5d10f63a894fd251R297-R299

Reply:
resolved in below thread